### PR TITLE
test: [M3-8725] - Add cypress tests for creating LKE clusters with ACL (LKE ACL integration part 3)

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
Adds tests for updating an LKE cluster's ACL from the summary/details page
Part 1: #10968 
Part 2: #11131
Part 3: #11132 

## Changes  🔄
- tests added:
  - Confirming ACL doesn't exist without the corresponding account capability
  - Confirming creating LKE cluster with ACL disabled
  - Confirming creating LKE cluster with ACL enabled
  - Confirming error/validation handling
- add related mock intercepts
- add relevant factories (some files will cause merge conflicts with part 2, #11131 - will fix as needed)

## Target release date 🗓️
10/28

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/kubernetes/lke-create.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Comment by coliu-akamai** [comment]
note - the copies are adding clutter to this test. I've created M3-8776 to clean that up and write unit tests for that instead, might combine that ticket with this one tbh! (just want to get to M3-8674 first)

File: packages/manager/cypress/e2e/core/kubernetes/lke-create.spec.ts
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/kubernetes/lke-create.spec.ts:1]: Question: should we be seeing the ACL flow visible in the pre-existing tests in this file? In LKE update spec we have the `apl` feature flag off in a beforeEach statement.

Test case: 'create an LKE cluster'
Screenshot 2024-10-23 at 12 29 48 PM

Comment (coliu-akamai) [reply packages/manager/cypress/e2e/core/kubernetes/lke-create.spec.ts:1]: yup so the apl feature flag controls APL, a different feature than ACL (I know, the acronyms are too similar 😆)

ACL is controlled by the account capability 'LKE Network Access Control List (IP ACL)' and there is no associated feature flag. Talmai had mentioned that the ACL capability is enabled as of last week to all customers - so ACL will be going out live to everyone on 10/28

Comment (mjac0bs) [reply packages/manager/cypress/e2e/core/kubernetes/lke-create.spec.ts:1]: 😵‍💫  I had a feeling I was getting things confused between APL and ACL. Wow. Okay, thank you for clarifying!

**Review by mjac0bs — APPROVED** [review]
Test coverage passes locally and coverage of cases looks good. Thanks for the follow up ticket you created. Did not see any typos here.

I do have a minor complaint about strange wording of the copy, but that's probably best for another time. 
`When enabled, only the IP addresses and ranges specified by you can connect to the control plane.` (passive voice)
⬇️ 
`When enabled, only the IP addresses and ranges you specify can connect to the control plane.` (active voice - clearer, more direct)

**Review by dwiley-akamai — APPROVED** [review]

**Comment by coliu-akamai** [comment]
going to be merging this as is, and will work on M3-8776 hopefully soon 🙏 
@mjac0bs will bring up your copy question to the LKE ACL team!

**Pull request merged**